app/services/karma_calculator.rb

```diff
@@ -7,9 +7,13 @@ def initialize(user)
   end
 
   def perform
-    user.karma.total = 0
-    return if user.created_at.blank?
+    if user.karma
+      user.karma.total = 0
+    else
+      user.karma = Karma.find_or_create_by(user_id: user.id, total: 0)
+    end
 
```

Comment:
@tansaku I would like to leave it as-is for now, since I tested it.  @mtc2013 can refactor.